projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-row.component.ts

```diff
@@ -31,36 +33,63 @@ export class IgxPivotRowComponent extends IgxRowDirective<IgxPivotGridComponent>
     /**
      * @hidden @internal
      */
-     @ViewChild('headerTemplate', { read: TemplateRef, static: true })
-     public headerTemplate: TemplateRef<any>;
+    @ViewChild('headerTemplate', { read: TemplateRef, static: true })
+    public headerTemplate: TemplateRef<any>;
 
     /**
      * @hidden @internal
      */
-     @ViewChild('headerTemplateGray', { read: TemplateRef, static: true })
-     public headerTemplateGray: TemplateRef<any>;
+    @ViewChild('headerTemplateGray', { read: TemplateRef, static: true })
+    public headerTemplateGray: TemplateRef<any>;
+
+    /**
+     * @hidden @internal
+     */
+    @ViewChild('headerDefaultTemplate', { read: TemplateRef, static: true })
+    public headerDefaultTemplate: TemplateRef<any>;
 
     public rowDimensionData: IPivotDimensionData[] = [];
 
     public get rowDimension() {
         return this.rowDimensionData?.map(x => x.column);
     }
 
+    /**
+     * @hidden
+     */
+    @Input()
+    @HostBinding('attr.aria-selected')
+    public get selected(): boolean {
+        let isSelected = false;
+        this.rowDimensionData.forEach(x => {
+            const key = this.getRowDimensionKey(x.column);
+            if (this.selectionService.isPivotRowSelected(key)) {
+                isSelected = true;
+                if (!this.grid.selectedPivotKeys.get(key)) {
+                    this.grid.selectedPivotKeys.set(key, this.rowData);
```

Comment:
Updating/setting collection in a getter does not seem ideal. Maybe this can be moved in selectPivotRow?

Reply:
In selectPivotRow we do not know exactly which siblings are related to the selected one. So when the getter is invoked for each row it`s only then we can populate the collection.